gxa/src/test/java/uk/ac/ebi/atlas/experimentpage/differential/DifferentialPathwaysComparisonServiceTest.java

```diff
@@ -0,0 +1,247 @@
+package uk.ac.ebi.atlas.experimentpage.differential;
+
+import com.google.common.collect.*;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.ac.ebi.atlas.experimentpage.context.DifferentialRequestContext;
+import uk.ac.ebi.atlas.experimentpage.context.DifferentialRequestContextFactory;
+import uk.ac.ebi.atlas.model.AssayGroup;
+import uk.ac.ebi.atlas.model.experiment.differential.*;
+import uk.ac.ebi.atlas.model.experiment.differential.microarray.MicroarrayExperimentConfiguration;
+import uk.ac.ebi.atlas.model.experiment.differential.microarray.MicroarrayExpression;
+import uk.ac.ebi.atlas.model.experiment.differential.microarray.MicroarrayProfile;
+import uk.ac.ebi.atlas.resource.MockDataFileHub;
+import uk.ac.ebi.atlas.trader.ConfigurationTrader;
+import uk.ac.ebi.atlas.web.DifferentialRequestPreferences;
+
+import java.util.*;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DifferentialPathwaysComparisonServiceTest {
+
+    @Mock
+    private DifferentialRequestContextFactory mockDifferentialRequestContextFactory;
+    @Mock
+    private DifferentialRequestContext mockRequestContext;
+    @Mock
+    private DifferentialRequestPreferences mockDifferentialRequestPreferences;
+    @Mock
+    private ConfigurationTrader mockConfigurationTrader;
+    @Mock
+    private MicroarrayExperimentConfiguration mockMicroarrayExperimentConfiguration;
+    @Mock
+    private DifferentialProfilesHeatMap mockDifferentialProfilesHeatMap;
+    @Mock
+    private DifferentialExperiment mockDifferentialExperiment;
+
+    private DifferentialPathwaysComparisonService subject;
+
+    private AssayGroup referenceAssay1 = new AssayGroup("g1", "assay1");
+    private AssayGroup testAssay1 = new AssayGroup("g2", "assay2");
+    private AssayGroup referenceAssay2 = new AssayGroup("g3", "assay3");
+    private AssayGroup testAssay2 = new AssayGroup("g4", "assay41", "assay42");
+
+    private Contrast c1 = new Contrast("g3_g1", null, referenceAssay1, testAssay1, "first contrast");
+    private Contrast c2 = new Contrast("g3_g2", null, referenceAssay2, testAssay2, "second contrast");
+
+    private String[] c1s1 = new String[]{"Term", "Accession", "Genes (tot)", "Stat (non-dir.) p", "p adj (non-dir.)"};
```

Comment:
This is hard to read. Doesn't matter now, but you could have also:
List<String[]> file1 = ImmutableList.of(
new String[]{"R-MMU-109581", "Apoptosis", "45", "0.535169358108118", "1"},
new String[]{"R-MMU-109581", "Apoptosis", "45", "0.535169358108118", "2"},
...
);

Reply:
Yes sure, I could have done it that way. 
I suppose this is a matter of programming styles, but there is nothing wrong with creating the strings separately and later add them in the empty list